### DISCUSSION TITLE
E2SM RC xapp_oran_slice_ctrl

### DISCUSSION THREAD

**Discussion by vamaglog** [srsran/srsRAN_Project#903] [General Help] [answered]
Hello,

I am trying to run the xapp_oran_slice_ctrl (https://gitlab.eurecom.fr/mosaic5g/flexric/-/blob/br-flexric/examples/xApp/c/control/xapp_oran_slice_ctrl.c) using the srsRAN_Project gNB, srsRAN_4G UE, open5GS and FlexRIC.

The UE is connected to the gNB via ZMQ and the gNB to the FlexRIC and to the core network, and I can send both DL and UL traffic. When I run the xapp_oran_slice_ctrl xApp, both the FlexRIC and the xAPP crash showing the following messages:

flexRIC:
![Screenshot 2024-10-28 at 12 40 08](https://github.com/user-attachments/assets/03e2930f-a56b-4f3e-a488-69f876d5d9e7)

xApp:
![Screenshot 2024-10-28 at 12 39 49](https://github.com/user-attachments/assets/364cbd73-b384-474f-bfc1-c4136c9e368c)

The log file from the gNB is the following:

![Screenshot 2024-10-28 at 12 43 31](https://github.com/user-attachments/assets/b5d96ffc-edde-4769-8c25-1761eda9fa03)

As you see, the logs say that "RAN function ID not supported".

Could you please assist me in debugging the problem?
Thank you very much in advance.
Vasilis

**Comment by pgawlowicz** [comment]
Could you share pcap file with the E2 Control Request message sent by FlexRIC?

**Reply by vamaglog** [reply]
Hi Piotr,

Here is the E2AP pcap file as requested.
[gnb_e2ap.pcap.zip](https://github.com/user-attachments/files/17629615/gnb_e2ap.pcap.zip)

**Comment by pgawlowicz** [comment]
@vamaglog It seems that you do not have e2sm-rc module enabled in the gnb config.

**Reply by vamaglog** [reply]
you are right! I enabled the module, however I am still facing a different issue this time. Below you may see more details at the screenshots below:

flexric:
![ric](https://github.com/user-attachments/assets/61bd9cbc-2ab0-42ab-a9b6-50dd5b7b4906)

xApp:
![xapp](https://github.com/user-attachments/assets/c5b06e42-5412-4b4f-bb95-b0f09e9de8a4)

gNB:
![gnb](https://github.com/user-attachments/assets/36ed4520-aced-4858-b335-a8202e97b5bf)

As you can see, the gNB crashes and as a result, the pcap file is corrupted. Wireshark says the following message: "The capture file appears to have been cut short in the middle of a packet".
In the zip file attached, I have provided the logs of the gNB in debug mode and the corrupted pcap file.

[logs.zip](https://github.com/user-attachments/files/17633493/logs.zip)

**Reply by pgawlowicz** [reply, marked as answer]
yes, there is an issue with PLMN parsing.  It will be fixed in the next release.

You disable (or simply avoid) PLMN parsing if you change its RAN-parameter-ID from 7 to 20 (for example) - [here](https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_rc/e2sm_rc_control_action_du_executor.cpp#L105C2-L105C46)

**Reply by vamaglog** [reply]
Hi Piotr,

Setting the PLMN param ID to 20 solved the issue. Thank you for the suggestion and support.

**Reply by pgawlowicz** [reply]
The issue will be fixed in the next project version.
